File: dataset_reduzido/avatar/Python/Code/atcoder_AGC043_B.py
def twos_in_factorial ( n ) :
    return n - bin ( n ).count ( '1' )
def parity_of_nCr ( n , i ) :
    f = twos_in_factorial
    return f ( n ) - f ( n - i ) - f ( i ) > 0
if __name__ == "__main__" :
    n = int ( input ( ) )
    nums = list ( map ( lambda x : int ( x ) - 1 , input ( ).strip ( ) ) )
    no_one = False
    if 1 not in nums :
        nums = [ num // 2 for num in nums ]
        no_one = True
    is_odd = False
    for i , num in enumerate ( nums ) :
        if not parity_of_nCr ( n - 1 , i ) and num & 1 :
            is_odd ^= 1
    if not is_odd :
        print ( 0 )
    else :
        print ( 1 << no_one )

File: output/LLaMa_repaired_1/avatar/Python/Java/atcoder_AGC043_B.java
import java.util.ArrayList;
import java.util.Scanner;

public class atcoder_AGC043_B {

    public static int twosInFactorial(int n) {
        String binary = Integer.toBinaryString(n);
        int count = 0;
        for (int i = 0; i < binary.length(); i++) {
            if (binary.charAt(i) == '1') {
                count++;
            }
        }
        return n - binary.length() - count;
    }

    public static boolean parityOfNCr(int n, int i) {
        int sum = 0;
        sum += twosInFactorial(n) - twosInFactorial(n - i) - twosInFactorial(i);
        return sum > 0;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());
        String[] input = scanner.nextLine().trim().split(" ");
        ArrayList<Integer> nums = new ArrayList<>();
        for (String num : input) {
            nums.add(Integer.parseInt(num) - 1);
        }
        boolean noOne = false;
        if (!nums.contains(0)) {
            ArrayList<Integer> newNums = new ArrayList<>();
            for (Integer num : nums) {
                newNums.add(num / 2);
            }
            nums = newNums;
            noOne = true;
        }
        boolean isOdd = false;
        for (int i = 0; i < nums.size(); i++) {
            if (!parityOfNCr(n - 1, i) && nums.get(i) % 2 != 0) {
                isOdd ^= true;
            }
        }
        if (!isOdd) {
            System.out.println(0);
        } else {
            System.out.println((noOne) ? 1 : 1 << 1);
        }
    }
}